sdks/python/apache_beam/transforms/core.py

```diff
@@ -2474,9 +2474,13 @@ def typed(transform):
       # Capture in closure (avoiding capturing self).
       args, kwargs = self.args, self.kwargs
 
+      #TODO: We need to add support for multi-window inputs to CombineGlobally()
       def inject_default(_, combined):
         if combined:
-          assert len(combined) == 1
+          if len(combined) > 1:
+            raise ValueError(
+                "Input from multiple simultaneous windows"
+                "isn't currently supported for CombineGlobally()")
```

Comment:
I chatted briefly with @robertwb and my understanding is that CombineGlobally is expected support multiple windows, so this message may be confusing to users. We'll need to look further to understand the failure mode here.

Reply:
My understanding is that, as defined, CombineGlobally is intended to return a single-element PCollection and I'm not sure how to interpret that in a multi-window context. My recollection is that attempting to implement an output with more than one element (e.g. "output = [1,2,3]"), as well as trying to nest that output in a list (e.g. "output = [[1,2,3]]"), both created broader issues for other parts of the execution.

As it relates to the issue at hand, CombineGlobally was only seeing multiple windows due to an issue with closing windows not triggering the transform correctly (see https://github.com/apache/beam/issues/24682) that has now been resolved. I also created https://github.com/apache/beam/issues/24683 to address this gap in CombineGlobally's functionality, since I felt it stretched beyond the scope of this issue.

Reply:
#24682 was fixed for PeriodicImpulse, I think #24438 was reproducibile w/ a PubSub source as well.

Reply:
If it can be repro'd w/ PubSub, that's also an issue that should be addressed. Do we need to do that to close the initial issue, or can that be its own issue?

Reply:
In my opinion the issue is the same, but I am not convinced we can close either at this time.

Reply:
Do we agree that this PR is at least a step in the right direction for now? I only set it to "address" #24438, so approving this PR won't close that issue.

Reply:
Apologies for slow response here. 

> My understanding is that, as defined, CombineGlobally is intended to return a single-element PCollection and I'm not sure how to interpret that in a multi-window context.

My understanding is that CombineGlobally should work for a multi-window context and result in combining down to 1 element  per each key+window pair. If you are interested in looking further into this problem, you can reproduce this pipeline in Java SDK, which presumably doesn't have this bug in examine the behavior, and/or get a second opinion. That being said,  in my opinion the "Input from multiple simultaneous windows isn't currently supported for CombineGlobally()" may discourage users from using CombineGlobally but it's some thing that we should fix instead.